ferveo-python/ferveo/__init__.pyi

```diff
@@ -190,3 +189,75 @@ def decrypt_with_shared_secret(
         dkg_params: DkgPublicParameters,
 ) -> bytes:
     ...
+
+
+class ThresholdEncryptionError(Exception):
+    pass
+
+
+class InvalidShareNumberParameter(Exception):
+    pass
+
+
+class InvalidDkgStateToDeal(Exception):
+    pass
+
+
+class InvalidDkgStateToAggregate(Exception):
+    pass
+
+
+class InvalidDkgStateToVerify(Exception):
+    pass
+
+
+class InvalidDkgStateToIngest(Exception):
+    pass
+
+
+class DealerNotInValidatorSet(Exception):
+    pass
+
+
+class UnknownDealer(Exception):
+    pass
+
+
+class DuplicateDealer(Exception):
+    pass
+
+
+class InvalidPvssTranscript(Exception):
+    pass
+
+
+class InsufficientTranscriptsForAggregate(Exception):
+    pass
+
+
+class InvalidFinalKey(Exception):
+    pass
+
+
+class InsufficientValidators(Exception):
+    pass
+
+
+class InvalidTranscriptAggregate(Exception):
+    pass
+
+
+class ValidatorsNotSorted(Exception):
+    pass
+
+
+class ValidatorPublicKeyMismatch(Exception):
+    pass
+
+
+class BincodeError(Exception):
+    pass
+
+
+class ArkSerializeError(Exception):
```

Comment:
This is a good point and on a similar note: How many typed errors do we need? Specifically, what are the cases where `nucypher` has to distinguish between Python exceptions in order to act on that information? Perhaps we only need a few o them, like `InvalidAggregate` and `ThresholdEncryptionError`, and everything we could put into `exception.ValueError` or something.

Reply:
Good question. Unsure at the moment. Happy to have a more in-depth discussion if you think it's worth exploring - defintely could use @KPrasch here.

That being said, perhaps we could also think about the best inheritance for these classes - eg. is it possible for some exceptions to subclass `ValueError` instead of `Exception` where appropriate eg. `InvalidShareNumberParameter` or `InvalidPvssTranscript`? Perhaps other logical sub-class groupings make sense. 

Then it can potentially be as general or granular as we like in Python - we either check for the superclass or the more specific exception type as needed. The granular work is already done in ferveo in this PR.